app/src/js/components/BatchAsyncCommands/BatchAsyncCommands.js

```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
```

Comment:
useEffect, from my understanding, helps with updating and fetching data but because it runs during every render, useRef, useCallback, and useMemo are hooks that can help depending on the situation (I didn't use it in this specific component but I used useCallback in a different ticket). -- this hook was a little bit tricky for me to understand and implement because of the rendering issue that it gives when used alone without other hooks. 

useState is used in place of this.setState in class methods- it just, quite literally, sets or initializes the state of a variable.

useRef allows for the storing or maintaining of values between renders which ends up helping the issue with re-rending by preventing it.